packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -588,26 +610,247 @@ export class UiControlClient extends ApiCommands {
+      throw new ValidationError(
+        '"relation" must never be undefined when searching for a switch or checkbox.',
+      );
     }
+    // param.relation.text must never be undefined here -> User error
+    if (element.relation.text === undefined) {
+      throw new ValidationError(
+        '"relation.text" must never be undefined when searching for a switch or checkbox.',
+      );
+    }
+    return this.evaluateRelation(
+      command,
+      element.relation.type,
+      element.relation.text,
+    ) as FluentFiltersOrRelationsGetter;
+  }
 
-    await command.exec();
+  /**
+   * Check if one or multiple elements are detected.
+   *
+   * **Examples:**
+   * ```typescript
+   * await aui.expectExistence([
+   *   {
+   *     type: 'text',
+   *     label: 'Switch to Dark',
+   *     matching: 'similar'
+   *   },
+   * ]);
+   *
+   * // Check for existence of multiple elements
+   * await aui.expectExistence([
+   *   {
+   *     type: 'textfield',
+   *     label: '42',
+   *     relation: {
+   *       type: 'below',
+   *       text: 'Register'
+   *     }
+   *   },
+   *   {
+   *     type: 'element',
+   *     label: 'Switch to Dark',
+   *   },
+   * ]);
+   *
+   * // Validate existence
+   * const exists = await aui.expectExistence([...]);
+   * exists.everythingExists // true when every element exists
+   *
+   * // Check which elements do not exist
+   * // with the elements property
+   * const nonExistentElements = exists.elements.filter((e) => e.exists===false)
+   * ```
+   *
+   * @param {ExpectExistenceInputParameter[]} params - Objects containing the required properties
+   *                                                   'type', 'label', 'matching' and 'relation'.
+   * @property {string} params.type - The type of the element: 'text' | 'button' | 'checkbox' |
+   *                                                           'switch' | 'textfield' | 'element'
+   * @property {string} params.label - The label text for the element.
+   *                                   In case of 'text': The text to be matched for
+   * @property {string} params.matching - Whether the text is matched using similarity,
+   *                                      exact match or a regular expression.
+   * @property {Object} params.relation - Object describing the relationship between the
+   *                                      clicked text and another element.
+   * @property {RelationsForConvenienceMethods} params.relation.type - The type of relation.
+   * @property {string} params.relation.text - The label or text associated with the
+   *                                           related element or state.
+   * @returns {ExpectExistenceReturnValue} - Property boolean 'everythingExists' tells
+   *                                         if every element exists.
+   */
+  async expectExistence(
+    params: ExpectExistenceInputParameter[],
+  ): Promise<ExpectExistenceReturnValue> {
+    if (params === undefined) {
```

Comment:
The error is not matching the condition. You are testing if `params` is `undefined` but you according to the error message, you seem like you want to be testing if params is an empty array. 

For the test if this is undefined, you could rely on typescript (cannot be undefined). If you want to test it as the user may not use typescript, I would have two conditions with a different validation error message for each.